lib/tasks/snapshots.rake

```diff
@@ -0,0 +1,34 @@
+require_relative '../../test/unit/foreman/renderer/template_snapshot_service'
+
+desc 'Snapshots tasks'
+namespace :snapshots do
+  desc "Generate snapshots"
+  task :generate => :environment do
+    unless Rails.env.test?
+      puts "This task can only be run in test environment"
+      exit
+    end
+
+    ENV['FIXTURES'] = 'settings,architectures,media,operatingsystems,taxonomies,subnets,environments,users,compute_resources,realms,images,hostgroups'
+    Rake::Task['db:reset'].invoke
+    Rake::Task['db:fixtures:load'].invoke
+
+    SETTINGS[:locations_enabled] = false
+    SETTINGS[:organizations_enabled] = false
+
+    host = TemplateSnapshotService.host
+    snapshots_directory = TemplateSnapshotService::SNAPSHOTS_DIRECTORY
+
+    TemplateSnapshotService.templates.each do |template|
+      path = File.join(snapshots_directory, template.template_kind.try(:name) || 'snippet', "#{template.name.tr(' ', '_')}.snap")
```

Comment:
I'd probably go for `snap.txt` extension, I can imagine that *.snap files will get opened in weird stuff on Windows or MacOS.